test/Core.Test/Services/Fido2AuthenticatorGetAssertionTests.cs

```diff
@@ -148,38 +155,61 @@ public async Task GetAssertionAsync_AsksForDiscoverableCredentials_ParamsDoesNot
         {
             // Arrange
-            _params.RequireUserVerification = true;
+            _params.UserVerificationPreference = Fido2UserVerificationPreference.Required;
             _userInterface.PickCredentialAsync(Arg.Any<Fido2GetAssertionUserInterfaceCredential[]>()).Returns((_ciphers[0].Id, true));
 
             // Act
             await _sutProvider.Sut.GetAssertionAsync(_params, _userInterface);
 
             // Assert
             await _userInterface.Received().PickCredentialAsync(Arg.Is<Fido2GetAssertionUserInterfaceCredential[]>(
-                (credentials) => credentials.All((c) => c.RequireUserVerification == true)
+                (credentials) => credentials.All((c) => c.UserVerificationPreference == Fido2UserVerificationPreference.Required)
+            ));
+        }
+
+        [Fact]
+        // Spec: Prompt the user to select a public key credential source `selectedCredential` from `credentialOptions`.
+        //       If UserVerificationPreference is Preferred and MP reprompt is on then the authorization gesture MUST include user verification.
+        //       If MP reprompt is off then the authorization gestue MAY include user verification
+        public async Task GetAssertionAsync_RequestsPreferredUserVerification_ParamsPreferUserVerification()
+        {
+            // Arrange
+            _params.UserVerificationPreference = Fido2UserVerificationPreference.Preferred;
+            _userInterface.PickCredentialAsync(Arg.Any<Fido2GetAssertionUserInterfaceCredential[]>()).Returns((_ciphers[0].Id, true));
+
+            // Act
+            await _sutProvider.Sut.GetAssertionAsync(_params, _userInterface);
+
+            // Assert
+            await _userInterface.Received().PickCredentialAsync(Arg.Is<Fido2GetAssertionUserInterfaceCredential[]>(
+                (credentials) => credentials.Any((c) => _ciphers.First(cip => cip.Id == c.CipherId).Reprompt == CipherRepromptType.None && c.UserVerificationPreference == Fido2UserVerificationPreference.Preferred)
+            ));
+
+            await _userInterface.Received().PickCredentialAsync(Arg.Is<Fido2GetAssertionUserInterfaceCredential[]>(
+                (credentials) => credentials.Any((c) => _ciphers.First(cip => cip.Id == c.CipherId).Reprompt != CipherRepromptType.None && c.UserVerificationPreference == Fido2UserVerificationPreference.Required)
             ));
         }
 
         [Fact]
         // Spec: Prompt the user to select a public key credential source `selectedCredential` from `credentialOptions`.
-        //       If `requireUserPresence` is true, the authorization gesture MUST include a test of user presence.
+        //       If `UserVerificationPreference` is Fido2UserVerificationPreference.Discouraged, the authorization gesture MUST include a test of user presence.
         // Comment: User presence is implied by the UI returning a credential.
         // Extension: UserVerification is required if the cipher requires reprompting.
```

Comment:
suggestion: the `Spec:` keyword uses copy-pasted lines from the specification so that you can google exactly which part of the algorithm that a certain test verifies. If we deviate from this then I suggest we write that on a separate row below

```diff
        // Spec: Prompt the user to select a public key credential source `selectedCredential` from `credentialOptions`.
-       //       If `UserVerificationPreference` is Fido2UserVerificationPreference.Discouraged, the authorization gesture MUST include a test of user presence.
+       //       If `requireUserPresence` is true, the authorization gesture MUST include a test of user presence. gesture MUST include a test of user presence.
        // Comment: User presence is implied by the UI returning a credential.
        // Extension: UserVerification is required if the cipher requires reprompting.
+       // Deviation: We send the actual preference instead of just a boolean, user presence is therefore required when that value is `discouraged`
```

Reply:
Alright, I guess you mean when that value is not `discouraged`, right? 
> ...user presence is therefore required when that value is `discouraged`

Reply:
Actually no, but maybe we could clarify this:

```
// Deviation: We send the actual preference instead of just a boolean, user presence (not user verification) is therefore required when that value is `discouraged`
```